### PR TITLE
small refactor and document update to support apollo-hook PR coming up

### PR DESCRIPTION
**Pull request by michelleyschen**
This is to prepare for https://github.com/Shopify/quilt/pull/642/files

Mostly I would like to publish the `react-async` changes, so I can start 🎩 `react-graphql`.
Right now I can't 🎩 properly since I changed two separate packages 😬

### CODE REVIEW

File: packages/react-testing/src/mount.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ type CustomMountResult<
   ? Promise<CustomRoot<Props, Context>>
   : CustomRoot<Props, Context>;
 
-class CustomRoot<Props, Context extends object> extends Root<Props> {
+export class CustomRoot<Props, Context extends object> extends Root<Props> {
```
Comment (michelleyschen) [inline packages/react-testing/src/mount.ts]: if I don't export this I get the following error

```
Exported variable 'mountWithGraphQL' has or is using name 'CustomRoot' from external module "Shopify/quilt/packages/react-testing/src/mount" but cannot be named.
```
Reason why
https://github.com/Microsoft/TypeScript/issues/5711

File: packages/react-testing/README.md [hunk 2]
```diff
@@ -387,7 +387,7 @@ export const mountWithGraphQL = createMount<Options, Context, true>({
 
     // Here's the important bit: resolve the GraphQL so our first queries are
     // in use for the component under test
-    await root.graphQL.resolveAll();
+    await root.context.graphQL.resolveAll();
```
Comment (michelleyschen) [inline packages/react-testing/README.md]: that `graphQL` is actually part of the `context` object, not `root`

File: packages/react-async/src/utilities.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const value = 'default' in module ? module.default : module;
+  return value == null ? null : value;
+}
+
+export async function resolve<T>(load: () => Promise<Import<T>>): Promise<T> {
+  try {
+    const resolved = await load();
+    return normalize(resolved);
+  } catch (error) {
+    throw Error('error loading module');
```
Comment (lemonmade) [inline packages/react-async/src/utilities.ts]: Should we bother with this or just let the original error through?

Comment (michelleyschen) [reply packages/react-async/src/utilities.ts]: ya I am fine with that too

**Pull request merged**